addons/l10n_sa_edi/data/ubl_21_zatca.xml

```diff
@@ -179,60 +209,158 @@
-                    <t t-call="{{InvoiceLineType_template}}">
-                        <t t-set="vals" t-value="foreach_vals"/>
-                    </t>
-                </cac:InvoiceLine>
+            <!-- Add Invoice Issue Time in compliance with rule KSA-25 -->
+            <xpath expr="//*[local-name()='IssueDate']" position="replace">
+                <t xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
+                    <cbc:IssueDate t-out="vals['issue_date'].strftime('%Y-%m-%d')"/>
+                    <cbc:IssueTime t-out="vals['issue_date'].strftime('%H:%M:%S')"/>
+                </t>
             </xpath>
-
-            <!-- Remove Order Reference if None -->
-            <xpath expr="//*[local-name()='OrderReference']" position="replace">
-                <cac:OrderReference t-if="vals.get('order_reference')"
-                                    xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2"
-                                    xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2"
-                                    xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
-                    <cbc:ID t-out="vals['order_reference']"/>
-                </cac:OrderReference>
+            <!-- Add Previous Invoice Hash & Invoice Counter Value -->
+            <xpath expr="//*[local-name()='BillingReference']" position="after">
+                <t xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2"
+                    xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2">
+                    <!-- Add QR Code in compliance with rule BR-KSA-27 -->
+                    <cac:AdditionalDocumentReference t-if="invoice._l10n_sa_is_simplified()">
+                        <cbc:ID>QR</cbc:ID>
+                        <cac:Attachment>
+                            <cbc:EmbeddedDocumentBinaryObject mimeCode="text/plain">N/A</cbc:EmbeddedDocumentBinaryObject>
+                        </cac:Attachment>
+                    </cac:AdditionalDocumentReference>
+                    <!-- Add Previous Invoice Hash in compliance with rule BR-KSA-61 -->
+                    <cac:AdditionalDocumentReference>
+                        <cbc:ID>PIH</cbc:ID>
+                        <cac:Attachment>
+                            <cbc:EmbeddedDocumentBinaryObject mimeCode="text/plain"
+                                t-out="vals.get('previous_invoice_hash')"/>
+                        </cac:Attachment>
+                    </cac:AdditionalDocumentReference>
+                    <!-- Add Invoice Counter Value in compliance with rules BR-KSA-33 and BR-KSA-34 -->
+                    <cac:AdditionalDocumentReference>
+                        <cbc:ID>ICV</cbc:ID>
+                        <cbc:UUID t-out="invoice.l10n_sa_chain_index"/>
+                    </cac:AdditionalDocumentReference>
```

Comment:
can't we add this in the UBL 2.0 template ? it's used by the CreditNote, DebitNote, and Invoice (http://www.datypic.com/sc/ubl20/e-cac_AdditionalDocumentReference.html)